[2020-12-22 14:20:27,255] {taskinstance.py:655} INFO - Dependencies all met for <TaskInstance: loan_prediction_train_model.preprocess_data 2020-12-22T12:43:23.210460+00:00 [queued]>
[2020-12-22 14:20:27,270] {taskinstance.py:655} INFO - Dependencies all met for <TaskInstance: loan_prediction_train_model.preprocess_data 2020-12-22T12:43:23.210460+00:00 [queued]>
[2020-12-22 14:20:27,270] {taskinstance.py:866} INFO - 
--------------------------------------------------------------------------------
[2020-12-22 14:20:27,270] {taskinstance.py:867} INFO - Starting attempt 47 of 48
[2020-12-22 14:20:27,271] {taskinstance.py:868} INFO - 
--------------------------------------------------------------------------------
[2020-12-22 14:20:27,296] {taskinstance.py:887} INFO - Executing <Task(LivyBatchOperator): preprocess_data> on 2020-12-22T12:43:23.210460+00:00
[2020-12-22 14:20:27,300] {standard_task_runner.py:53} INFO - Started process 61409 to run task
[2020-12-22 14:20:27,429] {logging_mixin.py:112} INFO - Running %s on host %s <TaskInstance: loan_prediction_train_model.preprocess_data 2020-12-22T12:43:23.210460+00:00 [running]> dcb43f18a24b
[2020-12-22 14:20:27,474] {logging_mixin.py:112} INFO - [2020-12-22 14:20:27,474] {batch.py:214} INFO - Submitting the batch to Livy... Payload:
{
  "file": "s3a://alanchn31-loan-default-prediction/dist/main.py",
  "args": [
    "--job",
    "preprocess_data",
    "--phase",
    "train",
    "--mode",
    "aws",
    "--awsKey",
    "AKIAUW4FKHTMR2UR3INH",
    "--awsSecretKey",
    "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",
    "--s3Bucket",
    "alanchn31-loan-default-prediction"
  ],
  "pyFiles": [
    "s3a://alanchn31-loan-default-prediction/dist/src.zip"
  ],
  "name": "preprocess_data_scheduled__2020-12-22T12:43:23.210460+00:00"
}
[2020-12-22 14:20:27,491] {logging_mixin.py:112} INFO - [2020-12-22 14:20:27,490] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:20:27,495] {logging_mixin.py:112} INFO - [2020-12-22 14:20:27,494] {http_hook.py:136} INFO - Sending 'POST' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches
[2020-12-22 14:20:27,553] {logging_mixin.py:112} INFO - [2020-12-22 14:20:27,553] {batch.py:166} INFO - Batch successfully submitted with id = 43.
[2020-12-22 14:20:27,554] {logging_mixin.py:112} INFO - [2020-12-22 14:20:27,553] {batch.py:80} INFO - Getting batch 43 status...
[2020-12-22 14:20:27,576] {logging_mixin.py:112} INFO - [2020-12-22 14:20:27,576] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:20:27,583] {logging_mixin.py:112} INFO - [2020-12-22 14:20:27,583] {http_hook.py:136} INFO - Sending 'GET' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/43
[2020-12-22 14:20:27,614] {logging_mixin.py:112} INFO - [2020-12-22 14:20:27,614] {batch.py:90} INFO - Batch 43 has not finished yet (state is 'starting')
[2020-12-22 14:20:47,638] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,638] {batch.py:80} INFO - Getting batch 43 status...
[2020-12-22 14:20:47,646] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,646] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:20:47,647] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,647] {http_hook.py:136} INFO - Sending 'GET' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/43
[2020-12-22 14:20:47,660] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,660] {batch.py:299} INFO - --------------------------------------------------Full log for batch 43--------------------------------------------------
[2020-12-22 14:20:47,671] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,671] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:20:47,673] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,672] {http_hook.py:136} INFO - Sending 'GET' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/43/log?from=0&size=100
[2020-12-22 14:20:47,687] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,687] {batch.py:309} INFO - stdout: 
[2020-12-22 14:20:47,689] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,687] {batch.py:309} INFO - 2020-12-22 14:20:29,741 WARN util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2020-12-22 14:20:47,690] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,689] {batch.py:309} INFO - Traceback (most recent call last):
[2020-12-22 14:20:47,690] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,690] {batch.py:309} INFO -   File "/tmp/spark-00985145-b69e-45d7-bdbd-f33905b4894f/main.py", line 87, in <module>
[2020-12-22 14:20:47,691] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,690] {batch.py:309} INFO -     main()
[2020-12-22 14:20:47,692] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,692] {batch.py:309} INFO -   File "/tmp/spark-00985145-b69e-45d7-bdbd-f33905b4894f/main.py", line 68, in main
[2020-12-22 14:20:47,693] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,692] {batch.py:309} INFO -     spark_logger.warn('no config file found')
[2020-12-22 14:20:47,694] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,693] {batch.py:309} INFO - NameError: name 'spark_logger' is not defined
[2020-12-22 14:20:47,694] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,694] {batch.py:309} INFO - 2020-12-22 14:20:34,681 INFO util.ShutdownHookManager: Shutdown hook called
[2020-12-22 14:20:47,694] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,694] {batch.py:309} INFO - 2020-12-22 14:20:34,683 INFO util.ShutdownHookManager: Deleting directory /tmp/spark-00985145-b69e-45d7-bdbd-f33905b4894f
[2020-12-22 14:20:47,694] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,694] {batch.py:309} INFO - 2020-12-22 14:20:34,688 INFO impl.MetricsSystemImpl: Stopping s3a-file-system metrics system...
[2020-12-22 14:20:47,694] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,694] {batch.py:309} INFO - 2020-12-22 14:20:34,688 INFO impl.MetricsSystemImpl: s3a-file-system metrics system stopped.
[2020-12-22 14:20:47,695] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,694] {batch.py:309} INFO - 2020-12-22 14:20:34,688 INFO impl.MetricsSystemImpl: s3a-file-system metrics system shutdown complete.
[2020-12-22 14:20:47,696] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,695] {batch.py:309} INFO - 
stderr: 
[2020-12-22 14:20:47,697] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,697] {batch.py:309} INFO - 
YARN Diagnostics: 
[2020-12-22 14:20:47,697] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,697] {batch.py:309} INFO - spark-submit start failed
[2020-12-22 14:20:47,701] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,698] {batch.py:318} INFO - --------------------------------------------------End of full log for batch 43--------------------------------------------------
[2020-12-22 14:20:47,701] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,701] {batch.py:335} INFO - Closing batch with id = 43
[2020-12-22 14:20:47,712] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,711] {base_hook.py:84} INFO - Using connection to: id: livy. Host: http://ec2-3-236-26-7.compute-1.amazonaws.com, Port: 8998, Schema: None, Login: None, Password: None, extra: None
[2020-12-22 14:20:47,712] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,712] {http_hook.py:136} INFO - Sending 'DELETE' to url: http://ec2-3-236-26-7.compute-1.amazonaws.com:8998/batches/43
[2020-12-22 14:20:47,727] {logging_mixin.py:112} INFO - [2020-12-22 14:20:47,726] {batch.py:340} INFO - Batch 43 has been closed
[2020-12-22 14:20:47,737] {taskinstance.py:1128} ERROR - Batch 43 failed with state 'dead'
Traceback (most recent call last):
  File "/opt/bitnami/python/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 966, in _run_raw_task
    result = task_copy.execute(context=context)
  File "/usr/local/airflow/plugins/airflow_livy/batch.py", line 173, in execute
    ).execute(context)
  File "/opt/bitnami/python/lib/python3.6/site-packages/airflow/sensors/base_sensor_operator.py", line 107, in execute
    while not self.poke(context):
  File "/usr/local/airflow/plugins/airflow_livy/batch.py", line 96, in poke
    raise AirflowException(f"Batch {self.batch_id} failed with state '{state}'")
airflow.exceptions.AirflowException: Batch 43 failed with state 'dead'
[2020-12-22 14:20:47,741] {taskinstance.py:1151} INFO - Marking task as UP_FOR_RETRY
[2020-12-22 14:20:52,302] {logging_mixin.py:112} INFO - [2020-12-22 14:20:52,302] {local_task_job.py:103} INFO - Task exited with return code 1
